### ISSUE TITLE
Android ListView - Restore Add/Remove item animations 

### ISSUE THREAD

**Issue by MatFillion** [unoplatform/uno#739]
## I'm submitting a...
- Regression (a behavior that used to work and stopped working in a new release) 

## Current behavior
Adding or removing items to ListView does not animate

## Expected behavior
Adding or removing items to ListView animates

## Minimal reproduction of the problem with instructions

Change feature flag `FeatureConfiguration.NativeListViewBase.RemoveItemAnimator` to false  (default is true) to compare behaviour change

## Environment

```
Nuget Package: 

Package Version(s): 

Affected platform(s):
- [ ] iOS
- [x] Android
- [ ] WebAssembly
- [ ] Windows
- [ ] Build tasks
```
Related to #702

**Comment by philipe-roberge** [comment]
I am facing this issue on Android right now. Here is the code I've written that should trigger the delete animation.

```c#
public ObservableCollection<Message> Messages { get; } = new ObservableCollection<Message>();

// list is populated...

public IDynamicCommand DeleteMessage => this.GetCommandFromTask<Message>(async (ct, message) =>
{
	await RunOnDispatcher(ct, async _ =>
	{
		if (await message.CanBeDeleted())
		{
			Messages.Remove(message); // This should trigger animation
		}
	});
});
```
The `Messages` are used in a `ListView` and everything works well on iOS and UWP, but our client is expecting the animation to work on Android as well, so this is a blocker for us.

**Comment by davidjohnoliver** [comment]
@philipe-roberge Does setting the flag mentioned in the issue help?

Ie if you add the following to your app startup (eg ctor of `App.xaml.cs`):

```csharp
#if __ANDROID__
    Uno.UI.FeatureConfiguration.NativeListViewBase.RemoveItemAnimator = false;
#endif
```

do you get item animations?

**Comment by philipe-roberge** [comment]
Yes, this did enable the animations for us. Was wondering if there was a reason this flag would be set to true by default?

**Comment by davidjohnoliver** [comment]
> Yes @davidjohnoliver, this did enable the animations for us. Was wondering if there was a reason this flag would be set to true by default?

It was disabled as a workaround for a crash that was occurring under particular circumstances: https://github.com/unoplatform/uno/pull/702

@Xiaoy312 may have more details.

**Comment by Xiaoy312** [comment]
The original crash was pretty elusive, I am having trouble reproducing it now.

We can restore flag for now, and see if the problem resurges somehow. It should be a safe bet.
If it comes back, we will have a repro and existing workaround. If not that will be perfect.

**Comment by Xiaoy312** [comment]
Currently removing this workaround causes a weird bug where
a ListView inside a StackPanel would not re-measure properly when new item(s) are added,
causing a ScrollViewer to appears instead of the ListView resizing when there is enough space to use.

This issue can be easily reproduced with `ListView_ObservableCollection_Unused_Space`, by adding this line to its .ctor:
```diff
public ListView_ObservableCollection_Unused_Space()
{
+    Uno.UI.FeatureConfiguration.NativeListViewBase.RemoveItemAnimator = false;
```

**Comment by jeromelaban** [comment]
@Xiaoy312 Is it something you can show in a video/screenshot ?

**Comment by MartinZikmund** [comment]
This is related to native, animations will have to be implemented in a general way for `ListView` on Skia targets
